### PR TITLE
Add more info and restructure migration guide

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: SDK v2 migration guide.md [hunk 1]
```diff
@@ -16,182 +16,189 @@ are outlines of the notable breaking changes as well as a mapping from version 1
...
-| `Message.ExpiryTimeUtc` | `TelemetryMessage.ExpiresOnUtc`, `IncomingMessage.ExpiresOnUtc` |
-| `MethodRequest` | `DirectMethodRequest` |
-| `MethodResponse` | `DirectMethodResponse` |
-| `DeviceAuthenticationWithRegistrySymmetricKey`, `DeviceAuthenticationWithSharedAccessPolicyKey`, `ModuleAuthenticationWithRegistrySymmetricKey` | `ClientAuthenticationWithSharedAccessKeyRefresh` |
-| `DeviceAuthenticationWithToken`, `ModuleAuthenticationWithToken` | `ClientAuthenticationWithSharedAccessSignature` |
-| `DeviceAuthenticationWithTokenRefresh`, `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithTokenRefresh` |
-| `DeviceAuthenticationWithX509Certificate` | `ClientAuthenticationWithX509Certificate` |
-
-#### Other notable breaking changes
+#### Notable breaking changes
```
Comment (drwill-ms) [inline SDK v2 migration guide.md:121]: Reordered to move the long table of text to the bottom of each client.

File: SDK v2 migration guide.md [hunk 2]
```diff
@@ -16,182 +16,189 @@ are outlines of the notable breaking changes as well as a mapping from version 1
...
 #### Notable additions
 
 - The device and module clients now have a property (e.g., `IotHubDeviceClient.ConnectionStatusInfo`) with the latest connection status information on it, eliminating the need for a connection status callback method to cache the latest values.
 - Added support for setting a client web socket instance in the client options so that users can have better control over AMQP web socket connections.
-- The library now includes IncrementalDelayRetryStrategy and FixedDelayRetryStrategy.
-- The client can now be re-opened after it has been closed. It cannot be re-opened after it has been disposed, though. Also, subscriptions do not carry over when the client is re-opened.
+- The library now includes 2 new `IRetryPolicy` implmentations: `IncrementalDelayRetryStrategy` and `FixedDelayRetryStrategy`.
+- The client can now be re-opened after it has been closed, removing the need to close/dispose/initialize/open; simply initialize/open, then if necessary close/open, and finally when done dispose.
+  > Be advised, subscriptions do not carry over when the client is re-opened.
+  >
```
Comment (brycewang-microsoft) [inline SDK v2 migration guide.md:149]: extra line?

File: SDK v2 migration guide.md [hunk 3]
```diff
@@ -16,182 +16,189 @@ are outlines of the notable breaking changes as well as a mapping from version 1
...
-|:---|:---|
-| `ModuleClient` | `IotHubModuleClient` |
-| `MessageResponse` | `MessageAcknowledgement` |
-| `Message` | `IncomingMessage`, `OutgoingMessage` |
+The device client and module client share a lot of API surface and underlying implementation. See changes to the IoT hub [device client](#deviceclient).
 
-#### Other notable breaking changes
+#### Notable breaking changes
 
-- See changes to `DeviceClient`.
```
Comment (brycewang-microsoft) [inline SDK v2 migration guide.md:139]: Why we are stating "N/A" instead of this?

File: SDK v2 migration guide.md (hunk 2 above)
Comment (drwill-ms) [reply SDK v2 migration guide.md:149]: Yes, it ensures the notes aren't all on the same line. A simple newline won't separate them.

File: SDK v2 migration guide.md (hunk 3 above)
Comment (drwill-ms) [reply SDK v2 migration guide.md:139]: See line 177 in the ne wfile.

File: SDK v2 migration guide.md [hunk 4]
```diff
@@ -16,182 +16,189 @@ are outlines of the notable breaking changes as well as a mapping from version 1
...
-- `RecommendedAction` enum provided in connection status handling based on connection status to assist in know what action the client app is recommended to take.
+The IoT hub device client now caches the latest connection status information so the client app does not have to do it. It can be found at `IotHubDeviceClient.ConnectionStatusInfo` (same property for module client). It is useful, for example, in avoiding starting an operation when the client is temporarily disconnected.
+
+In the connection status callback, a third value, `RecommendedAction`, is now provided to assist in knowing what action the client app is recommended to take. Previously, one would need to know based on every status and reason what action the client app should take, a nearly impossible task without reading a code sample.
+
+## What will happen to the v1 SDK
 
-## What will happen to the version 1 SDK
+After the v2 SDK is released as GA (general audience), meaning it is no longer only in preview, there will be one final LTS of the v1 SDK released that we will support like any other LTS release (security bug fixes, some non-security bug fixes as needed).
```
Comment (abhipsaMisra) [inline SDK v2 migration guide.md:107]: nit: isn't GA "general availability"?

**Review by brycewang-microsoft — APPROVED** [review]

File: SDK v2 migration guide.md [hunk 5]
```diff
@@ -16,182 +16,189 @@ are outlines of the notable breaking changes as well as a mapping from version 1
...
+  > We feel this caused more harm than good and as such it was best to remove.
 - Some options that were previously set in the `DeviceClient` constructor are now in the optional `IotHubClientOptions` parameter.
 - The connection status callback parameters have changed. Instead of two parameters, a class with several properties is provided.
-  - Two properties are the same as before, but with some renames (underscores removed) and obsolete members removed.
+  - Two properties enum values are mostly the same as before, with some renames (underscores removed) and obsolete members removed.
   - A new property has been added with a recommended action, which a device developer may observe or ignore.
 - The file upload method has been split into the three individual steps that this method used to take. See [this file upload sample](./iothub/device/samples/getting%20started/FileUploadSample/) for an example of how to do file upload using these discrete steps.
 - Cloud-to-device messages can be received by calling `SetMessageHandlerAsync` and providing a callback. Users no longer need to poll for messages with `ReceiveAsync`.
```
Comment (abhipsaMisra) [inline SDK v2 migration guide.md:134]: This is now "SetIncomingMessageCallbackAsync", right?

File: SDK v2 migration guide.md [hunk 6]
```diff
@@ -16,182 +16,189 @@ are outlines of the notable breaking changes as well as a mapping from version 1
...
-- Several callback handler set methods and definitions have changed, losing the `userContext` parameter.
+- Several callback handler set methods and definitions have changed, losing the `userContext` parameter which was deemed unnecessary and a vestige of the C device client.
```
Comment (abhipsaMisra) [inline SDK v2 migration guide.md:135]: Ah, I didn't know it was a C client pattern.

File: SDK v2 migration guide.md (hunk 3 above)
Comment (brycewang-microsoft) [reply SDK v2 migration guide.md:139]: My bad, I missed it.

Comment (abhipsaMisra) [reply SDK v2 migration guide.md:139]: How about we remove this header altogether? Someone could potentially skip the previous statement and think there were no breaking changes.

Comment (abhipsaMisra) [reply SDK v2 migration guide.md:139]: Or, "see above"?

File: SDK v2 migration guide.md (hunk 4 above)
Comment (drwill-ms) [reply SDK v2 migration guide.md:107]: Oops, yes.

**Review by abhipsaMisra — APPROVED** [review]

File: SDK v2 migration guide.md (hunk 6 above)
Comment (drwill-ms) [reply SDK v2 migration guide.md:135]: Pretty sure. I've seen that pattern for years in C, back before C# existed.

**Pull request merged**